modules/caas/api/src/main/java/io/cattle/platform/api/service/ServiceCreateValidationFilter.java

```diff
@@ -97,7 +98,7 @@ public Object create(String type, ApiRequest request, ResourceManager next) {
     public Object update(String type, String id, ApiRequest request, ResourceManager next) {
         Service service = objectManager.loadResource(Service.class, id);
 
-        validateLaunchConfigs(service, request);
+        request = validateLaunchConfigs(service, request);
```

Comment:
No need to return the request.  You can just modifying it in validationLaunchConfig